Activities/Database/UiPath.Database/DatabaseConnection.cs

```diff
@@ -94,6 +99,182 @@ public virtual int InsertDataTable(string tableName, DataTable dataTable)
             return dbDA.Update(dataTable);
         }
 
+        
+        public long BulkInsertDataTable(string tableName, DataTable dataTable, string connection)
+        {
+            DbDataAdapter dbDA = DbProviderFactories.GetFactory(_providerName).CreateDataAdapter();
+            DbCommandBuilder cmdb = DbProviderFactories.GetFactory(_providerName).CreateCommandBuilder();
+            cmdb.DataAdapter = dbDA;
+            dbDA.ContinueUpdateOnError = false;
+
+            long affectedRecords = 0;
+
+           
+
+            if (_providerName == "System.Data.SqlClient")
+            {
+                //if SQL Server
+
```

Comment:
we could extract code in separate methods. this body is pretty big

Reply:
method extracted separately